internal/db/params.go

```diff
@@ -60,12 +60,29 @@ func (db *Database) SaveCheckpointParams(
 		"type":    checkpointParamsType,
 		"version": checkpointParamsVersion, // hardcoded as 0
 	}
-	update := bson.M{"$setOnInsert": doc}
+	update := bson.M{"$set": doc}
 
 	_, err := collection.UpdateOne(ctx, filter, update, options.Update().SetUpsert(true))
 	return err
 }
 
+func (db *Database) GetCheckpointParams(ctx context.Context) (*bbnclient.CheckpointParams, error) {
```

Comment:
We only need this for tests, so I didn't include it in interface - no business logic use cases.